qcodes/dataset/measurements.py

```diff
@@ -37,14 +38,13 @@
 log = logging.getLogger(__name__)
 
 array_like_types = (tuple, list, np.ndarray)
-scalar_res_types = Union[str, int, float, np.dtype]
+scalar_res_types = Union[str, int, float, np.integer, np.floating]
```

Comment:
what about `np.complex` and `complex`?

Reply:
What about them :) They were not here so fixing that does not really belong in this pr but sure we can put them in

Reply:
> that does not really belong in this pr

agree

> so fixing

this is what i meant - whether the reason for them not being there is smth good or its a bug.

Reply:
It's a bug we do support them